### PR TITLE
Fix key delete crash on holding down

### PR DESCRIPTION
**Pull request by patrickgold**
Fix that the delete key crashes FlorisBoard when holding down on this key. Related to #97.

### CODE REVIEW

**Comment by patrickgold** [comment]
@RickyM7 could you please check if the debug build of this PR's build fixes the crash on your device? Thanks in advance!

**Comment by RickyM7** [comment]
The crash still occurs.

 ~~~ 1608913687019.stacktrace ~~~

android.view.ViewRootImpl$CalledFromWrongThreadException: Only the original thread that created a view hierarchy can touch its views.
	at android.view.ViewRootImpl.checkThread(ViewRootImpl.java:6956)
	at android.view.ViewRootImpl.invalidateChildInParent(ViewRootImpl.java:1119)
	at android.view.ViewGroup.invalidateChild(ViewGroup.java:5244)
	at android.view.View.invalidateInternal(View.java:13596)
	at android.view.View.invalidate(View.java:13532)
	at android.view.View.invalidateDrawable(View.java:17709)
	at android.graphics.drawable.Drawable.invalidateSelf(Drawable.java:432)
	at android.graphics.drawable.GradientDrawable.setTintList(GradientDrawable.java:1005)
	at android.view.View.applyBackgroundTint(View.java:18360)
	at android.view.View.setBackgroundTintList(View.java:18303)
	at dev.patrickgold.florisboard.util.View_utilsKt.setBackgroundTintColor2(view_utils.kt:38)
	at dev.patrickgold.florisboard.ime.text.key.KeyView.updateKeyPressedBackground(KeyView.kt:481)
	at dev.patrickgold.florisboard.ime.text.key.KeyView.setKeyPressed(KeyView.kt:61)
	at dev.patrickgold.florisboard.ime.text.key.KeyView.updateEnabledState(KeyView.kt:470)
	at dev.patrickgold.florisboard.ime.text.key.KeyView.updateVisibility(KeyView.kt:552)
	at dev.patrickgold.florisboard.ime.text.keyboard.KeyboardView.updateVisibility(KeyboardView.kt:361)
	at dev.patrickgold.florisboard.ime.text.smartbar.SmartbarView.updateSmartbarState(SmartbarView.kt:218)
	at dev.patrickgold.florisboard.ime.text.TextInputManager.sendKeyPress(TextInputManager.kt:666)
	at dev.patrickgold.florisboard.ime.text.key.KeyView$onFlorisTouchEvent$2.run(KeyView.kt:236)
	at java.util.TimerThread.mainLoop(Timer.java:555)
	at java.util.TimerThread.run(Timer.java:505)

**Comment by patrickgold** [comment]
Thanks for your fast response! It seems that I have to investigate this deeper than initially thought, as the crash occurs even when called from the Main thread, which is very strange.

**Comment by patrickgold** [comment]
@RickyM7 I've now finally found a virtual device where this crash was reproducible, allowing me to check and fix this issue. Just to confirm, could you please recheck if this crash is now finally resolved on your device too? Thanks!

**Comment by RickyM7** [comment]
@patrickgold Fixed on my phone. Remembering that my device is a Redmi Note 4 (MIUI 11 - Android 7.0). Thank you.

File: app/src/main/java/dev/patrickgold/florisboard/ime/text/key/KeyView.kt [hunk 1]
```diff
@@ -230,17 +235,15 @@ class KeyView(
...
+                    osTimer?.scheduleAtFixedRate(500, 50) {
+                        mainScope.launch(Dispatchers.Main) {
+                            florisboard?.textInputManager?.sendKeyPress(data)
                         }
-                    }, 500, 50)
+                        if (!isKeyPressed) {
+                            osTimer?.cancel()
+                            osTimer = null
+                        }
+                    }
```
Comment (yashx) [inline app/src/main/java/dev/patrickgold/florisboard/ime/text/key/KeyView.kt:246]: I think you will need to make this change [here](https://github.com/florisboard/florisboard/blob/e2c784f4cf8add029c2bdbb9be362530f7362bd3/app/src/main/java/dev/patrickgold/florisboard/ime/text/editing/EditingKeyView.kt#L103-L112) too

Comment (patrickgold) [reply app/src/main/java/dev/patrickgold/florisboard/ime/text/key/KeyView.kt:246]: Good catch, I will fix this immediately!

**Comment by patrickgold** [comment]
Thanks @RickyM7 and @yashx for the help in fixing this crash, I really appreciate it!

**Pull request merged**